qcodes/instrument_drivers/tektronix/Keithley_2450.py

```diff
@@ -0,0 +1,396 @@
+import numpy as np
+from typing import Any, Dict, Callable, cast
+
+from qcodes import VisaInstrument, InstrumentChannel, ArrayParameter
+from qcodes.instrument.parameter import Parameter, _BaseParameter
+from qcodes.utils.validators import Enum, Numbers
+
+
+class ParameterError(Exception):
+    """Raise this error if a parameter has an unexpected value"""
+    pass
+
+
+def assert_parameter_values(required_parameter_values: Dict[Parameter, Any]) -> None:
+    """
+    Assert that the parameters specified in the input dictionary have the required
+    values.
+
+    Args:
+        required_parameter_values
+    """
+    for parameter, required_value in required_parameter_values.items():
+
+        actual_value = parameter.get_latest() or parameter.get()
+        if required_value != actual_value:
+
+            instrument = cast(VisaInstrument, parameter.instrument)
+            instrument_name = instrument.name.replace("_", '.')
+
+            raise ParameterError(
+                f"'{instrument_name}.{parameter.name}' has value '{actual_value}'. "
+                f"This should be '{required_value}'"
+            )
+
+
+class Keithley2450Sweep(ArrayParameter):
```

Comment:
Is there any good reason for still using ArrayParameter. ParameterWithSetpoints has landed and is preferred

Reply:
Jens, 

I tried to locate documentation of `ParameterWithSetpoints` but I have been unable to find anything. Could you please point me to it? In the mean while, I will look at the tests of  `ParameterWithSetpoints` to figure out how to use it. I assume I can register the parameter in a dataset like usual?

Reply:
this is a bit strange that you couldn't find it. Well, there has been a problem with the search functionality of the docs due to a bug in a newer sphinx version, so this could be the problem.

However, now everything works, and the top 3 links on this page http://qcodes.github.io/Qcodes/search.html?q=parameterwithsetpoints&check_keywords=yes&area=default should give a reasonable overview on how it works and what it does. Please, feel free to ask more questions about.